Terminal.Gui/Configuration/SettingsScope.cs

```diff
@@ -102,7 +102,7 @@ public class SettingsScope : Scope<SettingsScope>
     /// <param name="resourceName"></param>
     public SettingsScope? UpdateFromResource (Assembly assembly, string resourceName)
     {
-        if ((resourceName == null) || string.IsNullOrEmpty (resourceName))
+        if (resourceName == null || string.IsNullOrEmpty (resourceName))
```

Comment:
Redundant and improper null check.

This should probably be the `ArgumentException.ThrowIfNullOrEmpty(resourceName);` helper both for brevity and because those helper methods actually have a run-time advantage over throwing it manually.

And then callers should be prepared to handle the exception.

Also, note that resources are one of the things that are potentially not AoT-friendly. I began looking into that a few days ago but life stuff got in the way and then I just haven't come back to that, yet, favoring working on outstanding work instead.